resources/migrations/001_update_migrations.yaml

```diff
@@ -5651,8 +5651,93 @@ databaseChangeLog:
             constraintName: fk_sandboxes_ref_permissions
             onDelete: CASCADE
 
+  #  -------------------------------------- Permissions migration ----------------------------------------------------
+  #  Think of the following migrations (v50.2024-02-27T13:45:*) as being one big chunk. On an upgrade, first we lock down
+  #  `permissions` so no *new* data permissions paths can be created anymore (e.g. in clustered metabase by another
+  #  instance running an old version of the code). Next we run the actual migration (moving data permissions from
+  #  `permissions` to `data_permissions`).
+  #
+  # On downgrade, we need to do things in the same order: first "unlock" permissions so data permissions paths can be
+  # created again, *then* move data permissions back to `permissions` from `data_permissions`. So we can't put the
+  # `rollback` for our permissions constraints on the same changeSet that created them, because they need to be rolled
+  # back *first*.
+  - changeSet:
+      id: v50.2024-02-27T20:51:15
+      author: johnswanson
+      comment: Create a stored procedure in MySQL to delete a constraint iff it exists
```

Comment:
MySQL has no built-in way to delete a constraint if and only if it exists. But we need to remove the constraint anytime we are inserting data that may violate it into the database - which means we need to make a way to do so. It turns out that the simplest way is with a stored procedure.